asserts/snap_asserts.go

```diff
@@ -698,3 +698,140 @@ func InitBuiltinBaseDeclaration(headers []byte) error {
+func (snapdev *SnapDeveloper) PublisherID() string {
+	return snapdev.HeaderString("publisher-id")
+}
+
+func (snapdev *SnapDeveloper) checkConsistency(db RODatabase, acck *AccountKey) error {
+	// Check authority is the publisher or trusted.
+	authorityID := snapdev.AuthorityID()
+	publisherID := snapdev.PublisherID()
+	if !db.IsTrustedAccount(authorityID) && (publisherID != authorityID) {
+		return fmt.Errorf("snap-developer must be signed by the publisher or a trusted authority but got authority %q and publisher %q", authorityID, publisherID)
+	}
+
+	_, err := db.Find(SnapDeclarationType, map[string]string{
+		// XXX: mediate getting current series through some context object? this gets the job done for now
+		"series":  release.Series,
+		"snap-id": snapdev.SnapID(),
+	})
+	if err == ErrNotFound {
+		return fmt.Errorf("snap-developer assertion for snap id %q does not have a matching snap-declaration assertion", snapdev.SnapID())
+	}
+
+	// check there's an account for the publisher-id
+	_, err = db.Find(AccountType, map[string]string{"account-id": publisherID})
+	if err == ErrNotFound {
+		return fmt.Errorf("snap-developer assertion for snap-id %q does not have a matching account assertion for the publisher %q", snapdev.SnapID(), publisherID)
+	}
+
+	// check there's an account for each developer
+	for developerID := range snapdev.developerRanges {
+		if developerID == publisherID {
+			continue
+		}
+		_, err = db.Find(AccountType, map[string]string{"account-id": developerID})
+		if err == ErrNotFound {
+			return fmt.Errorf("snap-developer assertion for snap-id %q does not have a matching account assertion for the developer %q", snapdev.SnapID(), developerID)
+		}
+	}
+
+	return nil
+}
+
+// sanity
+var _ consistencyChecker = (*SnapDeveloper)(nil)
+
+// Prerequisites returns references to this snap-developer's prerequisite assertions.
+func (snapdev *SnapDeveloper) Prerequisites() []*Ref {
+	// Capacity for the snap-declation, the publisher and all developers.
+	refs := make([]*Ref, 0, 2+len(snapdev.developerRanges))
+
+	// snap-declaration
+	// XXX: mediate getting current series through some context object? this gets the job done for now
+	refs = append(refs, &Ref{SnapDeclarationType, []string{release.Series, snapdev.SnapID()}})
+
+	// the publisher and developers
+	publisherID := snapdev.PublisherID()
+	refs = append(refs, &Ref{AccountType, []string{publisherID}})
+	for developerID := range snapdev.developerRanges {
+		if developerID != publisherID {
+			refs = append(refs, &Ref{AccountType, []string{developerID}})
+		}
+	}
+
+	return refs
+}
+
+func assembleSnapDeveloper(assert assertionBase) (Assertion, error) {
+	developerRanges, err := checkDevelopers(assert.headers, "developers")
+	if err != nil {
+		return nil, err
+	}
+
+	return &SnapDeveloper{
+		assertionBase:   assert,
+		developerRanges: developerRanges,
+	}, nil
+}
+
+func checkDevelopers(headers map[string]interface{}, name string) (map[string][]*dateRange, error) {
```

Comment:
I think here it would be ok to hardcode name for now?  until we plan to have this thing in another assertion